### PR TITLE
Objects overriding GetHashCode also override Equals

### PR DESCRIPTION
**Pull request by keegan-caruso**
Minor cleanup to AttributeKey

### CODE REVIEW

**Review by mafurman — APPROVED** [review]

**Comment by RojaEnnam** [comment]
Just curious. Did a sonar flag prompt for this change?

**Review by RojaEnnam — APPROVED** [review]

**Comment by keegan-caruso** [comment]
> Just curious. Did a sonar flag prompt for this change?

Yes, the dashboard should show it

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAttributeKeyComparer.cs [hunk 1]
```diff
@@ -70,56 +68,57 @@ public AttributeKey(SamlAttribute attribute)
...
+                hashCode ^= ValueType.GetHashCode();
+                hashCode ^= OriginalIssuer.GetHashCode();
+                return hashCode;
             }
 
+            /// <inheritdoc/>
+            public override int GetHashCode() => _hashCode;
+
+            /// <inheritdoc/>
+            public override bool Equals(object obj) => Equals(obj as AttributeKey);
```
Comment (GeoK) [inline src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAttributeKeyComparer.cs:87]: Was this override flagged too or you added it for correctness?

**Review by GeoK — APPROVED** [review]

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAttributeKeyComparer.cs (hunk 1 above)
Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAttributeKeyComparer.cs:87]: https://sonarcloud.io/project/issues?id=AzureAD_azure-activedirectory-identitymodel-extensions-for-dotnet&open=AXNs_zquUUMedrjWA3Kq&resolutions=WONTFIX&types=BUG

Yes, these were originally marked as won't fix, but that isn't correct as it effects collection behavior

**Pull request merged**